DataCollector/CommandDataCollector.php

```diff
@@ -29,11 +30,21 @@ public function collect(Request $request, Response $response, ?Throwable $except
         $this->data = [
             'num_commands' => count($this->commandLogger),
             'commands' => array_map(
-                static function (Command $command) : string {
-                    return json_encode($command->getCommand());
+                static function (Command $command) : array {
+                    return [
+                        'command' => json_encode($command->getCommand()),
```

Comment:
Maybe we can use `$this->cloneVar()` to create a vardumper clone and perform json encoding on the template instead? We could add the raw dump (`profiler_dump()`) of the command there and pretty-print the JSON too.

Reply:
Using `profiler_dump` will add a dependency to WebProfilerBundle. So, I just moved JSON encoding to template.

Maybe we could add a MongoDB query formatter later in order to display commands like this : `db.User.find({ _id: 1 });`

Reply:
@Jean-Beru I don't think adding a dependency is necessary for that. [DoctrineBundle](https://github.com/doctrine/DoctrineBundle/blob/2.4.x/composer.json) references `symfony/web-profiler-bundle` in `require-dev` and `suggest` only. This is how most bundles that use DataCollectors behave.

Reply:
@Jean-Beru we won't be able to format these commands into anything other than `db.runCommand(...)`, which is why I haven't done so. Furthermore, all cursor-involving commands are not retryable, so we shouldn't encourage users to actually run these commands.

As for the output, `json_encode` is fine, but since MongoDB uses BSON for its command, it may be more appropriate to change this to:
```php
use function MongoDB\BSON\fromPHP;
use function MongoDB\BSON\toCanonicalExtendedJSON;
// ...
return [
    'command' => toCanonicalExtendedJSON(fromPHP($command->getCommand()),
];
```

Since performance is not an issue, we don't need to worry about converting the PHP object back into a BSON string and then converting that to our [Extended JSON format](https://github.com/mongodb/specifications/blob/master/source/extended-json.rst). This preserves type information and is understood by all MongoDB drivers.

For now, I'm fine leaving the current output format in place, but this could be an improvement for future versions to let users choose whether they want plain json or extended json for use in other drivers.

Reply:
@alcaeus I added extended JSON format. Maybe we have to add `ext-mongodb` to requirement since they depend on `MongoDB\BSON` namespace ?

Reply:
Yep, we should. Depending on `ext-mongodb: ^1.0` would be the right constraint here.

Reply:
Done :wink: